Bottom: 937901f6ce5418610def565eb3b95f64b585c87e
Top:    17a3c2bb9c09764c778527f4cb326e6228dbbcd9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-30 01:02:37 +0000

Refresh of playing-with-stack-rendering

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index da6da72..819f882 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1054,11 +1054,12 @@ SetPVColor_inlayer (PinTypePtr Pin, LayerTypePtr Layer, int Type)
     color = PCB->ConnectedColor;
   else
     {
-      int component = GetLayerGroupNumberByNumber (max_layer + COMPONENT_LAYER);
-      int solder    = GetLayerGroupNumberByNumber (max_layer + SOLDER_LAYER);
+      int component_group = GetLayerGroupNumberByNumber (max_layer + COMPONENT_LAYER);
+      int solder_group    = GetLayerGroupNumberByNumber (max_layer + SOLDER_LAYER);
+      int this_group      = GetLayerGroupNumberByPointer (Layer);
 
-      if (Layer == LAYER_PTR (component) || Layer == LAYER_PTR (solder))
-        color = (SWAP_IDENT == (Layer == LAYER_PTR (solder)))
+      if (this_group == component_group || this_group == solder_group)
+        color = (SWAP_IDENT == (this_group == solder_group))
                   ? PCB->ViaColor : PCB->InvisibleObjectsColor;
       else
         color = Layer->Color;
